provider/consumer.py

```diff
@@ -278,8 +287,20 @@ def __pollForMessages(self):
                 if (message is not None):
                     if not message.error():
                         logging.debug("Consumed message: {}".format(str(message)))
-                        messageSize += len(message.value())
-                        messages.append(message)
+                        messageSize = self.__sizeMessage(message)
+                        if totalPayloadSize + messageSize > payload_limit:
+                            if len(messages) == 0:
+                                logging.error('[{}] Single message at offset {} exceeds payload size limit. Skipping this message!'.format(self.trigger, message.offset()))
```

Comment:
We don't currently have a nice mechanism for this. We are toying with the idea of exposing the trigger health info to the user or, even logs related to their triggers. That will be handled separately along with info related to automatic trigger disablement/deletion.